DeadmansSwitch.sol

```diff
@@ -5,28 +5,43 @@ contract DeadmansSwitch {
     // TODO: Declare state variables
     // Hint: You'll need variables for the owner, beneficiary, and last check-in block
 
+    address public owner;
+    address public beneficiary;
+    uint public lastCheckInBlock;
+
+
+            
     // TODO: Implement constructor
     constructor(address _beneficiary) {
-        // Hint: Initialize state variables
+        owner = msg.sender;
+        beneficiary= _beneficiary;
+        lastCheckInBlock=getCurrentBlock();
+
     }
 
     // TODO: Implement still_alive function
-    function still_alive() public {
-        // Hint: Update the last check-in block
+    function still_alive() public  {
+        require(owner==msg.sender);
+        lastCheckInBlock= getCurrentBlock();
     }
 
     // TODO: Implement release funds function
-    function releaseFunds() public {
+    function releaseFunds(address payable _beneficiary, address _owner) public  {
         // Hint: Check if 10 blocks have passed since last check-in
         // If so, transfer the contract balance to the beneficiary
+         require(block.number > (lastCheckInBlock + 10));
+         _beneficiary.transfer(_owner.balance);
```

Comment:
The contract should transfer its own balance to the beneficiary, not the owner's balance.